dali/pipeline/operators/reader/loader/video_loader.cc

```diff
@@ -217,9 +230,18 @@ OpenFile& VideoLoader::get_or_open_file(const std::string &filename) {
                                   stream->avg_frame_rate.num};
 
     // This check is based on heuristic FFMPEG API
+    AVPacket pkt = AVPacket{};
+    int ret;
+    while ((ret = av_read_frame(file.fmt_ctx_.get(), &pkt)) >= 0) {
```

Comment:
I started to wonder, how costly this check can get? Maybe we should provide it as an option? Something like `allow_fractional_frame_rate`? And normally just omit it and print an error with info that in fact it might be fractional frame rate and user should try that? Or maybe the performance penalty is not that high? Just a loose thought.

Reply:
For all the videos that I tested with the video frame is either the first frame or the second frame. So this while loop will probably only run once or twice in majority of videos. We break when we find the first video frame.

>Something like allow_fractional_frame_rate?

The reason i dont want to hide this check behind a flag is because for example NTSC uses 29.97 fps which is common. The ucf-101 dataset has videos with this fps. 
I also ran benchmark on about 1200 files which showed no significant differences between the two heuristics. 

```
-------------------------------------------------------------------------------------------------------------------
Benchmark                                                            Time           CPU Iterations UserCounters...
-------------------------------------------------------------------------------------------------------------------
VideoReaderBench/VideoPipe/2/1/4/9/iterations:100/real_time         62 ms          0 ms        100 FPS=81.5763/s

-------------------------------------------------------------------------------------------------------------------
Benchmark                                                            Time           CPU Iterations UserCounters...
-------------------------------------------------------------------------------------------------------------------
VideoReaderBench/VideoPipe/2/1/4/9/iterations:100/real_time         62 ms          0 ms        100 FPS=81.545/s

```

An alternative I can imagine is to combine it with the previous heuristic, When previous heuristic thinks that the video is VFR then we can try to verify again with this new heuristic.
But I can also imagine a video which looks like constant frame rate to the previous heuristic but is actually vfr.

What's your opinion ?

Reply:
For all the videos that I tested with the video frame is either the first frame or the second frame. So this while loop will probably only run once or twice in majority of videos. We break when we find the first video frame.

>Something like allow_fractional_frame_rate?

The reason i dont want to hide this check behind a flag is because for example NTSC uses 29.97 fps which is common. The ucf-101 dataset has videos with this fps.
I also ran benchmark on about 1200 files which showed no significant differences between the two heuristics. But this could be biased.
```
-------------------------------------------------------------------------------------------------------------------
Benchmark                                                            Time           CPU Iterations UserCounters...
-------------------------------------------------------------------------------------------------------------------
VideoReaderBench/VideoPipe/2/1/4/9/iterations:100/real_time         62 ms          0 ms        100 FPS=81.5763/s

-------------------------------------------------------------------------------------------------------------------
Benchmark                                                            Time           CPU Iterations UserCounters...
-------------------------------------------------------------------------------------------------------------------
VideoReaderBench/VideoPipe/2/1/4/9/iterations:100/real_time         62 ms          0 ms        100 FPS=81.545/s
```
An alternative I can imagine is to combine it with the previous heuristic, When previous heuristic thinks that the video is VFR then we can try to verify again with this new heuristic.
But I can also imagine a video which looks like constant frame rate to the previous heuristic but is actually vfr.

What's your opinion ?

Reply:
I think we should leave it as it is now. You are right. It's more robust to errors and issues as it is now.